pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/utils/TypeUtils.java

```diff
@@ -66,6 +66,12 @@ public static Object convert(Object value, ColumnDataType storedType) {
         if (value instanceof FloatArrayList) {
           // For ArrayAggregationFunction
           return ((FloatArrayList) value).elements();
+        } else if (value instanceof double[]) {
```

Comment:
Why do we need this special handling?

Reply:
Yes, `[0.1, 0.2, 0.3]` is parsed as `FLOAT_ARRAY` but internal representation is a `double[]`.

Reply:
Add some comment explaining this